packages/fscomponents/src/components/ModalHalfScreen.tsx

```diff
@@ -1,6 +1,8 @@
 import React, { PureComponent } from 'react';
 import {
   Animated,
+  Dimensions,
```

Comment:
We should avoid using Dimensions any time before componentDidMount is called, if possible, because it can mess up server-side rendering, where screen dimensions are not available.

Reply:
@Cauldrath So, as i understand we can use Dimensions after componentDidMount called?

Reply:
Correct. So you could set the height in state to a default value in the constructor and then grab it from Dimensions in componentDidMount. You should also check for updates to it, in case the user resizes a browser window or rotates a mobile/tablet device. Here's an example of setting up the Dimension listener:
https://github.com/brandingbrand/cvs.fs/blob/develop/packages/otchs-web/src/screens/Shop.tsx#L50

If you could get it to work with percentage-based positioning, that would avoid all that, though.

Reply:
Good think about percent positioning. I’ll check for state animation work with it (coz not sure), and will create one more commit.